nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -138,37 +141,37 @@ public WorkflowInstanceDao(SQLVariants sqlVariants, @NFlow JdbcTemplate nflowJdb
     workflowInstanceQueryMaxActionsDefault = env.getRequiredProperty("nflow.workflow.instance.query.max.actions.default",
         Long.class);
-    disableBatchUpdates = env.getRequiredProperty("nflow.db.disable_batch_updates", Boolean.class);
-    if (disableBatchUpdates) {
+    disableBatchUpdates.set(env.getRequiredProperty("nflow.db.disable_batch_updates", Boolean.class));
+    if (disableBatchUpdates.get()) {
       logger.info("nFlow DB batch updates are disabled (system property nflow.db.disable_batch_updates=true)");
     }
     workflowInstanceTypeCacheSize = env.getRequiredProperty("nflow.db.workflowInstanceType.cacheSize", Integer.class);
-    // In one deployment, FirstColumnLengthExtractor returned 0 column length (H2), so allow explicit length setting.
-    instanceStateTextLength = env.getProperty("nflow.workflow.instance.state.text.length", Integer.class, -1);
-    actionStateTextLength = env.getProperty("nflow.workflow.action.state.text.length", Integer.class, -1);
-    stateVariableValueMaxLength = env.getProperty("nflow.workflow.state.variable.value.length", Integer.class, -1);
+    instanceStateTextLength.set(env.getProperty("nflow.workflow.instance.state.text.length", Integer.class, -1));
+    actionStateTextLength.set(env.getProperty("nflow.workflow.action.state.text.length", Integer.class, -1));
+    stateVariableValueMaxLength.set(env.getProperty("nflow.workflow.state.variable.value.length", Integer.class, -1));
   }
 
   private int getInstanceStateTextLength() {
-    if (instanceStateTextLength == -1) {
-      instanceStateTextLength = jdbc.query("select state_text from nflow_workflow where 1 = 0", firstColumnLengthExtractor);
-    }
-    return instanceStateTextLength;
+    return getFieldLength(instanceStateTextLength, "state_text", WORKFLOW, "nflow.workflow.instance.state.text.length");
   }
 
-  int getActionStateTextLength() {
-    if (actionStateTextLength == -1) {
-      actionStateTextLength = jdbc.query("select state_text from nflow_workflow_action where 1 = 0", firstColumnLengthExtractor);
-    }
-    return actionStateTextLength;
+  private int getActionStateTextLength() {
+    return getFieldLength(actionStateTextLength, "state_text", ACTION, "nflow.workflow.action.state.text.length");
   }
 
   int getStateVariableValueMaxLength() {
-    if (stateVariableValueMaxLength == -1) {
-      stateVariableValueMaxLength = jdbc.query("select state_value from nflow_workflow_state where 1 = 0",
-          firstColumnLengthExtractor);
+    return getFieldLength(stateVariableValueMaxLength, "state_value", STATE, "nflow.workflow.state.variable.value.length");
+  }
+
+  private int getFieldLength(AtomicInteger length, String field, NflowTable table, String property) {
+    int value = length.get();
+    if (value == -1) {
+      String sql = new StringBuilder("select ").append(field).append(" from ").append(table.main).append(" where 1=0").toString();
```

Comment:
StringBuilder is slower than plain string concatenation here (where JVM will optimize it further).
=> Just add the strings together or even use String.format - perf does not mater.